### PR TITLE
Hide GitLab Details from Clients

### PR DESCRIPTION
**Pull request by dwasinge**
- created config resource/service to hide git details
- removed unused resources and services
- property name cleanup
- updated README

### CODE REVIEW

File: deployment/templates/gitlab-secret.yaml [hunk 1]
```diff
@@ -5,15 +5,11 @@ metadata:
...
   CONFIG_REPOSITORY_ID: "{{ .Values.configRepositoryId }}"
   DEPLOY_KEY: "{{ .Values.deployKey }}"
   GITLAB_API_URL: "{{ .Values.gitLabApiUrl }}"
   GITLAB_PERSONAL_ACCESS_TOKEN: "{{ .Values.gitLabPersonalAccessToken }}"
-  MP_JWT_VERIFY_PUBLICKEY_LOCATION: >-
-    {{ .Values.jwtVerifyPublicKeyLocation }}
-  RESIDENCIES_PARENT_REPOSITORIES_ID: "{{ .Values.residenciesParentRepositoryId }}"
-  TEMPLATE_REPOSITORY_ID: "{{ .Values.templateRepositoryId }}"
+  ENGAGEMENTS_REPOSITORY_ID: "{{ .Values.engagementsRepositoryId }}"
   TRUSTED_CLIENT_KEY: "{{ .Values.trustedClientKey }}"
```
Comment (mcanoy) [inline deployment/templates/gitlab-secret.yaml:14]: again.

File: deployment/values-dev.yaml [hunk 2]
```diff
@@ -13,11 +13,10 @@ git:
...
+configFile: false
 configRepositoryId: false
 deployKey: false
 gitLabApiUrl: false
 gitLabPersonalAccessToken: false
-jwtVerifyPublicKeyLocation: false
-residenciesParentRepositoryId: false
-templateRepositoryId: false
+engagementsRepositoryId: false
 trustedClientKey: false
```
Comment (mcanoy) [inline deployment/values-dev.yaml:22]: same

File: deployment/README.md [hunk 3]
```diff
@@ -31,13 +31,12 @@ helm template . \
...
+  --set configFile=<your-config-file> \
   --set configRepositoryId=<your-config-repository-id> \
   --set deployKey=<your-gitlab-deploy-key-id> \
   --set gitLabApiUrl=<your-gitlab-base-url> \
   --set gitLabPersonalAccessToken=<your-gitlab-personal-access-token> \
-  --set jwtVerifyPublicKeyLocation=<your-openid-connect-cert-url> \
-  --set residenciesParentRepositoryId=<your-gitlab-group-id> \
-  --set templateRepositoryId=<your-template-repository-id> \
+  --set engagementsRepositoryId=<your-gitlab-group-id> \
   --set trustedClientKey=<your-trusted-client-key> \
```
[the remark this answers is not in the document]
Comment (dwasinge) [reply deployment/README.md:40]: Just for my own clarification, I changed residenciesParentRepositoryId  to engagementsRepositoryId, which is being using in EngagementService to set the parent for the customer group.  engagementRepositoryId is being used in the EngagementService and configRepositoryId in the ConfigService.  I did remove the templateRepositoryId as it was the same as the configRepositoryId and it was no longer being used.

[the remark this answers is not in the document]
Comment (dwasinge) [reply deployment/README.md:40]: Or did you mean the trustedClientKey?  That one I don't think is being used that I could tell, just wasn't sure why it was there so I left it.

[the remark this answers is not in the document]
Comment (mcanoy) [reply deployment/README.md:40]: yeah. trustedClientKey

[the remark this answers is not in the document]
Comment (dwasinge) [reply deployment/README.md]: removed

File: deployment/templates/gitlab-secret.yaml (hunk 1 above)
Comment (dwasinge) [reply deployment/templates/gitlab-secret.yaml]: removed

File: deployment/values-dev.yaml (hunk 2 above)
Comment (dwasinge) [reply deployment/values-dev.yaml]: removed

**Pull request merged**